### ISSUE TITLE
[Feature request] Netplay

### ISSUE THREAD

**Issue by karasuhebi** [PCSX2/pcsx2#772]
PCSX2 has a DEV9 plugin for the network adapter which lets users play games online that originally had an online mode, but it's lacking any kind of netplay features so that users can play local multiplayer games online.

I know it probably won't be easy to do (keeping emulators in sync over the internet and such) but I'd love to see PCSX2 get netplay at some point. The PS2 has some great games and it'd be fun to be able to play them with my online friends some day. :)

**Comment by refractionpcsx2** [comment]
We tried contacting the author of PCSX2 Online to get his changes implemented but he never responded to us, I think most devs have little interest in doing this currently.

**Comment by karasuhebi** [comment]
So what should this issue be labeled as? Is there a "Back burner" label? XD

**Comment by refractionpcsx2** [comment]
how about that? :P

**Comment by karasuhebi** [comment]
Looks good to me. XD Thanks!

**Comment by dan83sama** [comment]
https://github.com/alexsharoff/pcsx2-online

**Comment by refractionpcsx2** [comment]
That's hugely out of date, merging the online stuff from that now would be a complete nightmare.

**Comment by ProhorovIlja** [comment]
There is a branch for a 1.7.0 dev version of PCSX2 with a netplay.
https://github.com/nipkownix/pcsx2-online
Maybe it is possible to implement it into a PCSX2.
Cuz PS2 netplay for a very few games that do support lan netplay is ok, but not as huge as a local netplay through hamachi, radmin vpn, etc.

**Comment by ThisNekoGuy** [comment]
I hope to God this can be done soon; Dolphin has had this feature for a long time already and I've been itching to play PS2 games with my friend for ages.

**Comment by refractionpcsx2** [comment]
This branch is 1519 commits behind PCSX2:master

I wouldn't call that a 1.7 build.

If they want to PR the changes, we can merge them in, I'm not against looking at it, but the code needs to be updated.

**Comment by tadanokojin** [comment]
> https://github.com/alexsharoff/pcsx2-online

This one doesn't have a commit dedicated to adding netplay, instead the init commit (including all of pcsx2's source code) adds it. [Kind of makes it hard to parse what is going on](https://github.com/alexsharoff/pcsx2-online/commit/0bfa3c50b124819c81bf9dd01260479e298175ed). Also has a boost dependency so that sucks.

> https://github.com/nipkownix/pcsx2-online

[Seems a bit more reasonable](https://github.com/nipkownix/pcsx2-online/commits/4ccdc5e89727491aa5f8d7cf90cd9fe264638737?after=4ccdc5e89727491aa5f8d7cf90cd9fe264638737+34&branch=4ccdc5e89727491aa5f8d7cf90cd9fe264638737&path%5B%5D=pcsx2&path%5B%5D=Netplay) but still kind of messy history wise.

Both projects seem pretty abandoned. The latter hasn't seen a commit or been responsive to issues in over year. Doubt we'll see much on that front but if someone wants to contact the author and see if they're interested in getting their work upstreamed that's fine. Otherwise some other party is going to have to go through the code and PR it. Not even how well it works. I have doubts it'll be as easy as including this code.

**Comment by vico93** [comment]
So... any news about this?

**Comment by BevP90** [comment]
My post was closed. Would be interested to hear what the PCSX2 team think of implementing a delay based netcode client.

**Comment by Mrlinkwii** [comment]
Pull requests welcome I suppose , the team work on what they want

**Comment by salimpam12** [comment]
I need play mortal combat saolin monks or basara heroes in 2 komputer, how? I play epcsx ps1 can play CTR use net play

**Comment by kokroo** [comment]
I believe apps like Parsec have largely replaced the need for this, although I understand it is not the same thing.

**Comment by roxas0zero** [comment]
any update on this ?

**Comment by ButterCCM** [comment]
Have save state loading and saving times been significantly improved? I know the plan broadly was for a delay based system but if we could achieve rollback or at the least have it as an option for computers fast enough it would be game changing. Alternatively if a delay system works well enough maybe we get a slippi like setup for Tekken 5 or something.

**Comment by Meowdoleon** [comment]
My main use case supporting this is for RetroAchievement.
Parsec is great and all, but only one of the player can get the achievement on Multiplayer allowed sets (such as https://retroachievements.org/game/19095). With Netplay, all players could get achievements.

**Comment by zander765trr-web** [comment]
Desperately wishing this was a thing, since there are no options for PCSX2 netplay on Linux. Parsec does not allow hosting on Linux and typical Steam Remote Play systems do not work with Flatpaks.

Please at least consider an option for us to netplay. It would be much appreciated by the community, I know it.

**Comment by AmandaRoseChaqueta** [comment]
> Desperately wishing this was a thing, since there are no options for PCSX2 netplay on Linux. Parsec does not allow hosting on Linux and typical Steam Remote Play systems do not work with Flatpaks.
> 
> Please at least consider an option for us to netplay. It would be much appreciated by the community, I know it.

The main issue is that there are no devs that know how to implement something like this, at least without being an absolute mess.
